src/main/java/mate/academy/bookstore/serivce/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.bookstore.serivce;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws EntityNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find user by email "
+                        + email));
+    }
+}
```

Comment:
Actually we have UsernameNotFoundException (How I know when you are overriding this method you get it by default), but I think it's okay